Migration of two tables at the same time one of the job prepare failed

## Bug Report

migration MySQL  tables 
```
mysql> CREATE MIGRATION PROCESS CONFIGURATION (
    -> READ(
    -> WORKER_THREAD=48,
    -> BATCH_SIZE=1000,
    -> SHARDING_SIZE=1000000
    -> ),
    -> WRITE(
    -> WORKER_THREAD=48,
    -> BATCH_SIZE=1000
    -> ),
    -> STREAM_CHANNEL (TYPE(NAME='MEMORY',PROPERTIES('block-queue-size'='10000')))
    -> );
Query OK, 0 rows affected (0.52 sec)

mysql> ADD MIGRATION SOURCE RESOURCE source_ds (
    ->     URL="jdbc:mysql://127.0.0.1:3306/scaling_bmsql?serverTimezone=UTC&useSSL=false",
    ->     USER="root",
    ->     PASSWORD="root",
    ->     PROPERTIES("maxPoolSize"="50","idleTimeout"="60000")
    -> );
Query OK, 0 rows affected (0.82 sec)

mysql> MIGRATE TABLE source_ds.bmsql_order_line INTO scaling_bmsql.bmsql_order_line;
Query OK, 0 rows affected (0.96 sec)

mysql> MIGRATE TABLE source_ds.bmsql_order_line_copy INTO scaling_bmsql.bmsql_order_line_copy;
Query OK, 0 rows affected (1.24 sec)
```

sometimes, the exception occur.
```
Exception in thread "Thread-4" org.apache.shardingsphere.data.pipeline.core.exception.PipelineJobPrepareFailedException: prepare target tables failed.
	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.prepareTargetTables(MySQLDataSourcePreparer.java:43)
	at org.apache.shardingsphere.data.pipeline.core.prepare.PipelineJobPreparerUtils.prepareTargetTables(PipelineJobPreparerUtils.java:98)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareTarget(MigrationJobPreparer.java:162)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareAndCheckTarget(MigrationJobPreparer.java:132)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepareAndCheckTargetWithLock(MigrationJobPreparer.java:116)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJobPreparer.prepare(MigrationJobPreparer.java:77)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJob.prepare(MigrationJob.java:92)
	at org.apache.shardingsphere.data.pipeline.scenario.migration.MigrationJob.lambda$execute$0(MigrationJob.java:82)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "repository" already exists; SQL statement:
CREATE TABLE `repository`(id varchar(36) PRIMARY KEY, `key` TEXT, `value` TEXT, parent TEXT) [42101-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:502)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.command.ddl.CreateTable.update(CreateTable.java:91)
	at org.h2.command.CommandContainer.update(CommandContainer.java:169)
	at org.h2.command.Command.executeUpdate(Command.java:252)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:252)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:223)
	at com.zaxxer.hikari.pool.ProxyStatement.execute(ProxyStatement.java:95)
	at com.zaxxer.hikari.pool.HikariProxyStatement.execute(HikariProxyStatement.java)
	at org.apache.shardingsphere.mode.repository.standalone.jdbc.JDBCRepository.init(JDBCRepository.java:68)
	at org.apache.shardingsphere.infra.util.spi.type.required.RequiredSPIRegistry.getRegisteredService(RequiredSPIRegistry.java:45)
	at org.apache.shardingsphere.mode.repository.standalone.StandalonePersistRepositoryFactory.getInstance(StandalonePersistRepositoryFactory.java:44)
	at org.apache.shardingsphere.mode.manager.standalone.StandaloneContextManagerBuilder.build(StandaloneContextManagerBuilder.java:47)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.createContextManager(ShardingSphereDataSource.java:76)
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.<init>(ShardingSphereDataSource.java:64)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:93)
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:152)
	at org.apache.shardingsphere.driver.data.pipeline.datasource.creator.ShardingSpherePipelineDataSourceCreator.createPipelineDataSource(ShardingSpherePipelineDataSourceCreator.java:51)
	at org.apache.shardingsphere.data.pipeline.core.datasource.PipelineDataSourceFactory.newInstance(PipelineDataSourceFactory.java:44)
	at org.apache.shardingsphere.data.pipeline.core.datasource.DefaultPipelineDataSourceManager.getDataSource(DefaultPipelineDataSourceManager.java:53)
	at org.apache.shardingsphere.data.pipeline.core.prepare.datasource.AbstractDataSourcePreparer.getCachedDataSource(AbstractDataSourcePreparer.java:99)
	at org.apache.shardingsphere.data.pipeline.mysql.prepare.datasource.MySQLDataSourcePreparer.prepareTargetTables(MySQLDataSourcePreparer.java:38)
```

### Which version of ShardingSphere did you use?

master 309c34fb64647c584bddfdc3cac9b0ec25853b85

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-Proxy

### Expected behavior

No exception thrown

### Actual behavior

Exception thrown and prepare failed.

### Reason analyze (If you can)

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-h2/src/main/java/org/apache/shardingsphere/mode/repository/standalone/h2/H2JDBCRepositoryProvider.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-standalone-mode/shardingsphere-standalone-mode-repository/shardingsphere-standalone-mode-repository-provider/shardingsphere-standalone-mode-repository-jdbc/shardingsphere-standalone-mode-repository-jdbc-h2/src/main/java/org/apache/shardingsphere/mode/repository/standalone/h2/H2JDBCRepositoryProvider.java
@@ -31,7 +31,7 @@ public final class H2JDBCRepositoryProvider implements JDBCRepositoryProvider {
     
     @Override
     public String createTableSQL() {
-        return "CREATE TABLE `repository`(id varchar(36) PRIMARY KEY, `key` TEXT, `value` TEXT, parent TEXT)";
+        return "CREATE TABLE IF NOT EXISTS `repository`(id varchar(36) PRIMARY KEY, `key` TEXT, `value` TEXT, parent TEXT)";
     }
     
     @Override

